pkg/components/init_job.go

```diff
@@ -56,7 +56,12 @@ func NewInitJob(
 	conditionsManager apiproxy.ConditionManager,
 	imagePullSecrets []corev1.LocalObjectReference,
 	name, configFileName, image string,
+	instanceImage *string,
```

Comment:
I think there is a wrong place/responsibility of the job object to know about two possible images and decide which one to use. Let's stick with one image argument in the job and pass the correct one from the outside (it will require some copy-paste or helper method, because it is the same logic in different components)

Reply:
Done